tests/unit/test_charm.py

```diff
@@ -137,120 +141,19 @@ def _set_other_node_api_address_in_peer_relation(self, relation_id: int, unit_na
             key_values=key_values,
         )
 
-    @patch("charm.generate_vault_unit_certificate")
-    @patch("ops.model.Model.get_binding")
     @patch("ops.model.Container.remove_path")
-    def test_given_not_leader_when_install_then_existing_data_is_removed(
```

Comment:
Is there any test which validates the behavior of removed test "not_leader_when_install" ?

Reply:
On install we don't care anymore whether the unit is leader or not so there's no need to test this specifically. Here's the on install handler:

```python
def _on_install(self, event: InstallEvent):
        """Handler triggered when the charm is installed.

        Sets pebble plan, initializes vault, enable audit device, and unseals vault.
        """
        if not self._container.can_connect():
            self.unit.status = WaitingStatus("Waiting to be able to connect to vault unit")
            event.defer()
            return
        self._delete_vault_data()
```